Sources/Private/Experimental/Animations/ShapeAnimation.swift

```diff
@@ -42,43 +42,17 @@ extension CAShapeLayer {
       context: context)
   }
 
-  /// Adds animations for properties related to the given `Stroke` object (`strokeColor`, `lineWidth`, etc)
-  @nonobjc
-  func addAnimations(for stroke: Stroke, context: LayerAnimationContext) {
-    lineJoin = stroke.lineJoin.caLineJoin
-    lineCap = stroke.lineCap.caLineCap
-    miterLimit = CGFloat(stroke.miterLimit)
-
-    addAnimation(
-      for: .strokeColor,
-      keyframes: stroke.color.keyframes,
-      value: \.cgColorValue,
-      context: context)
-
-    addAnimation(
-      for: .lineWidth,
-      keyframes: stroke.width.keyframes,
-      value: \.cgFloatValue,
-      context: context)
-
-    if let (dashPattern, dashPhase) = stroke.dashPattern?.shapeLayerConfiguration {
-      lineDashPattern = dashPattern.map {
-        KeyframeGroup(keyframes: $0).exactlyOneKeyframe.value.cgFloatValue as NSNumber
-      }
-
-      addAnimation(
-        for: .lineDashPhase,
-        keyframes: dashPhase,
-        value: \.cgFloatValue,
-        context: context)
-    }
-  }
-
   /// Adds animations for `strokeStart` and `strokeEnd` from the given `Trim` object
   @nonobjc
   func addAnimations(for trim: Trim, context: LayerAnimationContext) {
     let (strokeStartKeyframes, strokeEndKeyframes) = trim.caShapeLayerKeyframes()
 
+    if trim.offset.keyframes.contains(where: { $0.value.cgFloatValue != 0 }) {
+      LottieLogger.shared.assertionFailure("""
+      The CoreAnimation rendering engine doesn't support Trim offsets
```

Comment:
Some of the examples I was working with use `trim.offset`, which I hadn't encountered before. I investigated a bit and it seems pretty tough to support properly. 

In two examples I was able to work around this by removing the `trim.offset` from the animation json and just adding a corresponding `rotation`. That only works for circles, though, and may not generalize to all types of `offset` values 🤷🏼‍♂️